features/features.go

```diff
@@ -161,18 +164,40 @@ var featuresSupportedCallbacks = map[SnapdFeature]func() (bool, string){
 	},
 	// UserDaemons requires user units
 	UserDaemons: func() (bool, string) {
-		if !release.SystemctlSupportsUserUnits() {
+		if !releaseSystemctlSupportsUserUnits() {
 			return false, "user session daemons are not supported on this system's distribution version"
 		}
 		return true, ""
 	},
-	// AppArmorPrompting requires a newer version of snapd with all the
+	// AppArmorPrompting requires AppArmor parser and kernel support for
+	// prompting, as well as a newer version of snapd with all the
 	// prompting components in place. TODO: change this callback once ready.
 	AppArmorPrompting: func() (bool, string) {
+		kernelFeatures, err := apparmorKernelFeatures()
+		if err != nil {
+			return false, fmt.Sprintf("error checking apparmor kernel features: %v", err)
```

Comment:
nitpick: these could follow our usual error format (`cannot do xyz`) for the ones that are actually errors